src/backend/distributed/executor/multi_utility.c

```diff
@@ -1255,7 +1255,13 @@ ExecuteDistributedDDLCommand(Oid relationId, const char *ddlCommandString,
 {
 	bool executionOK = false;
 
-	PreventTransactionChain(isTopLevel, "distributed DDL commands");
+	if (IsInTransactionChain(isTopLevel) && (XactModification > XACT_MODIFICATION_NONE))
```

Comment:
Hm, taking it back. There are many other places we check the modification value and throw an error without checking transaction state, and the above logic for an `Assert` would apply there too. Removing the check altogether. Was probably being cautious as I wasn't as familiar with the DDL pipeline.